libminifi/test/integration/OnScheduleErrorHandlingTests.cpp

```diff
@@ -25,31 +25,45 @@
 #include "core/state/ProcessorController.h"
 #include "../TestBase.h"
 #include "../KamikazeProcessor.h"
+#include "utils/IntegrationTestUtils.h"
 
 /*Verify behavior in case exceptions are thrown in onSchedule or onTrigger functions
  * KamikazeProcessor is a test processor to trigger errors in these functions */
 class KamikazeErrorHandlingTests : public IntegrationBase {
  public:
   void runAssertions() override {
-    std::string logs = LogTestController::getInstance().log_output.str();
-
-    auto result = countPatInStr(logs, minifi::processors::KamikazeProcessor::OnScheduleExceptionStr);
-    size_t last_pos = result.first;
-    int occurrences = result.second;
-
-    assert(occurrences > 1);  // Verify retry of onSchedule and onUnSchedule calls
+    using org::apache::nifi::minifi::utils::verifyEventHappenedInPollTime;
+    assert(verifyEventHappenedInPollTime(std::chrono::milliseconds(wait_time_), [&] {
+      const std::string logs = LogTestController::getInstance().log_output.str();
+      const auto result = countPatInStr(logs, minifi::processors::KamikazeProcessor::OnScheduleExceptionStr);
+      const int occurrences = result.second;
+      if (occurrences <= 1) {  // Verify retry of onSchedule and onUnSchedule calls
```

Comment:
```return result.second > 1```
?

Reply:
I like naming this occurences, it will probably get optimized anyway, but I wouldn't know what it means otherwise. Also I almost always prefer relation operators `<` and `<=` to `>` and `>=`.

Reply:
You can usually choose to use `<`/`<=` without changing the semantics. But why does it matter, except when writing generic code that must only depend on `operator<`?
What about this?
```
const int occurrences = result.second;
return 1 < occurrences;
```

Or possibly change `countOccurrences` to return a struct instead of a pair.

Reply:
Updated according to the suggestion, I did not realize that the suggestion is actually combining the return statements. I completely agree that it is the sane thing to do, I do not know how I missed it :D

Updated the PublishKafkaOnScheduleTests as well that follows a similar pattern.